### PR TITLE
chore: remove manual memo calcs

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Use the `memo` value given to us by `thorchain/swap/quote` - there is no need to calculate it ourselves so long as we are providing the correct input data when fetching the quote.

Things still to check:

- [x] Slippage input amount when hitting the `GET` request
- [ ] Refund handling for small swaps - we might need client-side logic to handle what was previously a swapper concern

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Assists with the Swapper SDK work by simplifying the input data needed to fetch a quote.

## Risk

High - a bug with a THORSwap `memo` can lead to a total loss of trade funds. Test accordingly.

## Testing

Check:

- We still get THORSwap quotes as expected

### Engineering

Check:

- The memo returned from the `thorchain/swap/quote` endpoint matches what we expect (compare with that on prod if necessary with some inline debugging)

### Operations

Check:

- THORSwap trades complete as expected, which a memo that matches expectations

## Screenshots (if applicable)

N/A

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5094** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5094" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5094?utm_source=stack-comment).

**Review by woodenfurniture — APPROVED** [review]
amazing, you legend

File: src/lib/swapper/swappers/ThorchainSwapper/getThorTradeQuote/getTradeQuote.ts [hunk 1]
```diff
@@ -50,25 +48,18 @@ type ThorTradeQuote =
...
     sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmountCryptoBaseUnit,
-    slippageTolerancePercentage,
     accountNumber,
     chainId,
     receiveAddress,
     affiliateBps,
     wallet,
   } = input
 
-  const slippageTolerance =
```
Comment (0xApotheosis) [inline src/lib/swapper/swappers/ThorchainSwapper/getThorTradeQuote/getTradeQuote.ts:67]: TODO: put me in the quote request as `tolerance_bps`.

Comment (0xApotheosis) [reply src/lib/swapper/swappers/ThorchainSwapper/getThorTradeQuote/getTradeQuote.ts:67]: Nope, don't do this `tolerance_bps` isn't actually slippage. Calculate it manually (https://github.com/shapeshift/web/pull/5094/commits/c84faf2f1b2d974a1228dc33c41abea217ad08e9).

**Comment by 0xApotheosis** [comment]
This one could do with some good ol' fashioned @gomesalexandre scrutiny.

File: src/lib/swapper/swappers/ThorchainSwapper/endpoints.ts [hunk 2]
```diff
@@ -41,7 +41,9 @@ export const thorchainApi: Swapper2Api = {
       }))
     }
 
-    return await getThorTradeQuote(input, rates).then(async firstQuote => {
+    return await getThorTradeQuote({
+      ...input,
```
Comment (gomesalexandre) [inline src/lib/swapper/swappers/ThorchainSwapper/endpoints.ts:45]: Remove spread?

Comment (gomesalexandre) [reply src/lib/swapper/swappers/ThorchainSwapper/endpoints.ts]: 0dacf628d488b506f646236353b7b7db166a4191

**Review by gomesalexandre — COMMENTED** [review]
First pass, looking very good to me according to the updated test cases - functional pass to follow before making this purple

**Review by gomesalexandre — APPROVED** [review]
Tested against prod with RUNE <-> ATOM and RUNE -> DOGE:

### RUNE <-> ATOM

- "You Get" amount is showing no regressions against prod

image
image

- Memo is looking good against prod

image
image

And limit is slightly higher too 🎉 

- Trade is going through

image

### RUNE -> DOGE

- "You Get" amount is showing no regressions against prod

image
image

Note how we get slightly more here 🎉 

- Memo is looking good against prod

image
image

- Trade is going through

image

Notice how the receive amount is now higher, as a result of the higher limit 🎉

File: src/lib/swapper/swappers/ThorchainSwapper/endpoints.ts (hunk 2 above)
Comment (0xApotheosis) [reply src/lib/swapper/swappers/ThorchainSwapper/endpoints.ts]: Nice, not sure why that was there!

**Pull request merged**